daprdocs/content/en/developing-applications/building-blocks/state-management/howto-state-query-api.md

```diff
@@ -0,0 +1,140 @@
+| `AND`    | []operation|  operation[0] AND operation[1] AND ... AND operation[n] |
+| `OR`      | []operation|  operation[0] OR operation[1] OR ... OR operation[n] |
+
+The `sort` is an optional section is an ordered array of `key:order` pairs, where `key` is a key in the state store, and the `order` is an optional string indicating sorting order: `"ASC"` for ascending and `"DESC"` for descending. If omitted, ascending order is the default.
+
+The `pagination` is an optional section containing `limit` and `token` parameters. `limit` sets the page size. `token` is an iteration token returned by the component, and used in consecutive queries.
+
+Below is an example of the query request JSON body.
+
+This query retrieves all records where people either (1) belong to "Dev Ops" organization, or (2) belong to "Finance" and reside in Washington or California.
+```json
+{
+    "query": {
+        "filter": {
+            "OR": [
+                {
+                    "EQ": {"person.org": "Dev Ops"}
+                },
+                {
+                    "AND": [
+                        {
+                            "EQ": {"person.org": "Finance"}
+                        },
+                        {
+                            "IN": {"state": ["CA", "WA"]}
+                        }
+                    ]
+                }
+            ]
+        },
+        "sort": [
+            {
+            "key": "state",
+                "order": "DESC"
+            },
+            {
+                "key": "person.id"
+            }
+        ],
+        "pagination": {
+            "limit": 3
+        }
+    }
+}
+```
+
+The query request is translated into a native query language and executed in the state store component.
+
+In the case of SQL, the aforementioned query is translated into
+```sql
+SELECT * FROM c WHERE
+  person.org = "Dev Ops" OR
+  (person.org = "Finance" AND state IN ("CA", "WA"))
+ORDER BY
+  state DESC,
+  person.id ASC
+LIMIT 3
+```
+
+Upon successful execution, the state store returns a JSON object with a list of matching items and the token:
+```json
+{
+  "results": [
+    {
+      "key": "4",
+      "data": {
+        "person": {
+          "id": 1021,
+          "org": "Dev Ops"
+        },
+        "city": "Seattle",
+        "state": "WA"
+      },
+      "etag": "d58da615-2003-44ad-b349-cbabd54dcbfe"
+    },
+    {
+      "key": "1",
+      "data": {
+        "city": "Spokane",
+        "state": "WA",
+        "person": {
+          "org": "Dev Ops",
+          "id": 1034,
+        }
+      },
+      "etag": "256c951a-b63d-4c6c-a383-d2dc8f9d5071"
+    },
+    {
+      "key": "3",
+      "data": {
+        "person": {
+          "org": "Finance",
+          "id": 1017
+        },
+        "city": "Sacramento",
+        "state": "CA"
+      },
+      "etag": "704217a7-edc9-4b5c-b027-8468b84106a9"
+    }
+  ],
+  "token": "5"
+}
+```
```

Comment:
You need to add  an example that shows how to use the pagination token on a subsequent query as the final example.
This

The `pagination` is an optional section containing `limit` and `token` parameters. `limit` sets the page size. `token` is an iteration token returned by the component, and used in consecutive queries.